### PR TITLE
Filetype updates

### PR DESCRIPTION
**Pull request by gpanders**
Aggregation of filetype.lua updates

### CODE REVIEW

File: runtime/lua/vim/filetype.lua [hunk 1]
```diff
@@ -770,6 +770,14 @@ local extension = {
   mm = function() vim.fn["dist#ft#FTmm"]() end,
   mms = function() vim.fn["dist#ft#FTmms"]() end,
   p = function() vim.fn["dist#ft#FTprogress_pascal"]() end,
+  patch = function(path, bufnr)
+    local firstline = getline(bufnr, 1)
+    if string.find(firstline, "^From " .. string.rep("%x", 40) .. "+ Mon Sep 17 00:00:00 2001$") then
```
Comment (KillTheMule) [inline runtime/lua/vim/filetype.lua:775]: I've no idea what this is about, but the date somehow looks suspicious. Are you sure that's what should be here?

Comment (gpanders) [reply runtime/lua/vim/filetype.lua:775]: [Yes](https://github.com/neovim/neovim/blob/master/runtime/filetype.vim#L496)

Comment (KillTheMule) [reply runtime/lua/vim/filetype.lua:775]: Well then :)

**Comment by SanchayanMaity** [comment]
Can we merge this?

**Pull request merged**